design/authenticators/authn_oidc/oidc_configuration_enhancements.md

```diff
@@ -0,0 +1,228 @@
+to a couple other small gems which we need to consider as well:
+1. SWD (Simple Web Discovery) - handles OIDC configuration discovery against the provider URI configuration 
+endpoint: `/.well-known/openid-configuration` 
+1. WebFinger - used for dynamic discovery when the credential issuer is unknown to the client. **Not used in Conjur.**
+
+### Considerations around OIDC Discovery
+The SWD gem discussed previously is used by OpenIDConnect across many authenticators in Conjur (authn-oidc, 
+authn-jwt, authn-azure, authn-gcp). It performs a couple common tasks in Conjur by invoking the OIDC discovery 
+endpoint:
+  - validate authenticator status and provider connectivity
+  - fetch provider keys for decoding tokens
+
+This should not interfere with the proposed solution, but it is important to note since the modified code will be 
+used across different authenticators to invoke the provider discovery endpoint `/.well-known/openid-configuration`. 
+It will be necessary that the other authenticators are unaffected by this change.
+
+### Testing
+The testing will depend on the final implementation, but at the very least the wrapper method will need to be 
+well-covered by unit tests.
+
+#### Unit tests (temporary truststore modifications)
+1. OpenSSL truststore contains the CA cert symlink while yielding
+1. OpenSSL truststore symlink is removed after yielding
+1. Hash collision in truststore does not produce an error but instead increments the file extension of the symlink
+1. Temp file/dir exists while yielding
+1. Temp file/dir contains expected certificate content while yielding
+1. Temp file/dir is removed after yielding
+
+#### Unit tests (authenticator data object)
+1. Default `ca-cert` value is nil
+1. Set value matches expected
+
+#### Cucumber tests
+One or more of our Keycloak OIDC scenarios should be updated to use a `ca-cert` value provided via Conjur 
+variable. It may turn out to be easier to update all keycloak examples to use the cert variable, and the case 
+where the variable is not set can be covered by the Okta examples.
+
+### Open Questions
+- Do we need validation that `ca-cert` is a valid cert before attempting to use it?
+- Should debug logging of OIDC be enabled by default when Conjur is running in debug?
+- Should OpenSSL debug logs be included?
+
+## Tasks
+### 1. Update the OpenIDConnect gem (1 pt)
+Upgrade our OpenIDConnect gem to >= 2.0.0 (when it switched from `httpclient` to `faraday`) and ensure nothing 
+breaks.
+
+### 2. Create a temporary truststore wrapper method (2 pts)
+Create a method which accepts a string (`ca_cert`) and temporarily updates the default OpenSSL truststore, and 
+yields to another method. Add tests to ensure expected state during code execution and cleanup has occured 
+afterwards.
+
+I would suggest adding it to the [CertUtils module](https://github.com/cyberark/conjur/blob/master/app/domain/conjur/cert_utils.rb).
+
+### 3. Add support for `ca-cert` variable in authn-oidc config (2 pts)
```

Comment:
Is this the only visible user facing change needed to support ca-certs?

Reply:
For customers upgrading from v13.0, do they just need to add this variable in the auth policy for custom certs to work?

Reply:
Yes, the `ca-cert` variable will be optional and is the only UX change.